sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/trace/propagation/aws_xray_format.py

```diff
@@ -0,0 +1,283 @@
+            return trace.set_span_in_context(
+                trace.INVALID_SPAN, context=context
+            )
+
+        trace_id, span_id, sampled, err = self.extract_span_properties(
+            trace_header
+        )
+
+        if err is not None:
+            return trace.set_span_in_context(
+                trace.INVALID_SPAN, context=context
+            )
+
+        options = 0
+        if sampled:
+            options |= trace.TraceFlags.SAMPLED
+
+        span_context = trace.SpanContext(
+            trace_id=trace_id,
+            span_id=span_id,
+            is_remote=True,
+            trace_flags=trace.TraceFlags(options),
+            trace_state=trace.TraceState(),
+        )
+
+        if not span_context.is_valid:
+            _logger.error(
+                "Invalid Span Extracted. Insertting INVALID span into provided context."
+            )
+            return trace.set_span_in_context(
+                trace.INVALID_SPAN, context=context
+            )
+
+        return trace.set_span_in_context(
+            trace.DefaultSpan(span_context), context=context
+        )
+
+    def extract_span_properties(self, trace_header):
+        trace_id = trace.INVALID_TRACE_ID
+        span_id = trace.INVALID_SPAN_ID
+        sampled = False
+
+        extract_err = None
+
+        for kv_pair_str in trace_header.split(self.KV_PAIR_DELIMITER):
+            if extract_err:
+                break
+
+            try:
+                key_str, value_str = kv_pair_str.split(
+                    self.KEY_AND_VALUE_DELIMITER
+                )
+                key, value = key_str.strip(), value_str.strip()
+            except ValueError:
+                _logger.error(
+                    (
+                        "Error parsing X-Ray trace header. Invalid key value pair: %s. Returning INVALID span context.",
+                        kv_pair_str,
+                    )
+                )
+                return trace_id, span_id, sampled, extract_err
+
+            if key == self.TRACE_ID_KEY:
+                if not self.validate_trace_id(value):
+                    _logger.error(
+                        (
+                            "Invalid TraceId in X-Ray trace header: '%s' with value '%s'. Returning INVALID span context.",
+                            self.TRACE_HEADER_KEY,
+                            trace_header,
+                        )
+                    )
+                    extract_err = True
+                    break
+
+                try:
+                    trace_id = self.parse_trace_id(value)
+                except ValueError:
+                    _logger.error(
+                        (
+                            "Invalid TraceId in X-Ray trace header: '%s' with value '%s'. Returning INVALID span context.",
+                            self.TRACE_HEADER_KEY,
+                            trace_header,
+                        )
+                    )
+                    extract_err = True
```

Comment:
does this need a break?

Reply:
I don't think it _needed_ it but just helped it exit earlier, but now that we raise an exception it's not needed :)